examples/all-clusters-app/esp32/main/Rpc.cpp

```diff
@@ -0,0 +1,326 @@
+    { 132, 5660 }, { 134, 5670 }, { 136, 5680 }, { 138, 5690 }, { 140, 5700 }, { 142, 5710 }, { 144, 5720 }, { 149, 5745 },
+    { 151, 5755 }, { 153, 5765 }, { 155, 5775 }, { 157, 5785 }, { 159, 5795 }, { 161, 5805 }, { 165, 5825 }, { 169, 5845 },
+    { 173, 5865 }, { 183, 4915 }, { 184, 4920 }, { 185, 4925 }, { 187, 4935 }, { 188, 4940 }, { 189, 4945 }, { 192, 4960 },
+    { 196, 4980 },
+} });
+
+// These are potentially large objects for the scan results.
+constexpr size_t kScanRecordsMax = sizeof(chip_rpc_ScanResults().aps) / sizeof(chip_rpc_ScanResult);
+chip_rpc_ScanResults out_scan_records;
+wifi_ap_record_t scan_records[kScanRecordsMax];
+
+class Device final : public generated::Device<Device>
+{
+public:
+    pw::Status FactoryReset(ServerContext & ctx, const pw_protobuf_Empty & request, pw_protobuf_Empty & response)
+    {
+        chip::DeviceLayer::ConfigurationMgr().InitiateFactoryReset();
+        return pw::OkStatus();
+    }
+    pw::Status Reboot(ServerContext & ctx, const pw_protobuf_Empty & request, pw_protobuf_Empty & response)
+    {
+        return pw::Status::Unimplemented();
+    }
+    pw::Status TriggerOta(ServerContext & ctx, const pw_protobuf_Empty & request, pw_protobuf_Empty & response)
+    {
+        return pw::Status::Unimplemented();
+    }
+    pw::Status GetDeviceInfo(ServerContext &, const pw_protobuf_Empty & request, chip_rpc_DeviceInfo & response)
+    {
+        response.vendor_id        = 1234;
+        response.product_id       = 5678;
+        response.software_version = 0;
+        return pw::OkStatus();
+    }
+};
+
+class Button final : public generated::Button<Button>
+{
+public:
+    pw::Status Event(ServerContext &, const chip_rpc_ButtonEvent & request, pw_protobuf_Empty & response)
+    {
+#if CONFIG_DEVICE_TYPE_M5STACK
+        if (request.pushed)
+        {
+            ScreenManager::ButtonPressed(1 + request.idx);
+        }
+        return pw::OkStatus();
+#else  // CONFIG_DEVICE_TYPE_M5STACK
+        return pw::Status::Unimplemented();
+#endif // CONFIG_DEVICE_TYPE_M5STACK
+    }
+};
+
+class Wifi final : public generated::Wifi<Wifi>
+{
+public:
+    pw::Status GetChannel(ServerContext &, const pw_protobuf_Empty & request, chip_rpc_Channel & response)
+    {
+        uint8_t channel = 0;
+        wifi_second_chan_t second;
+        PW_TRY(EspToPwStatus(esp_wifi_get_channel(&channel, &second)));
+        response.channel = channel;
+        return pw::OkStatus();
+    }
+
+    pw::Status GetSsid(ServerContext &, const pw_protobuf_Empty & request, chip_rpc_Ssid & response)
+    {
+        wifi_config_t config;
+        PW_TRY(EspToPwStatus(esp_wifi_get_config(WIFI_IF_STA, &config)));
+        size_t size = std::min(sizeof(response.ssid.bytes), sizeof(config.sta.ssid));
+        memcpy(response.ssid.bytes, config.sta.ssid, sizeof(response.ssid.bytes));
+        response.ssid.size = size;
```

Comment:
For example, if `sizeof(response.ssid.bytes) > sizeof(config.sta.ssid)` this will read uninitialized memory, because the third arg to `memcpy` is not `size`.

Reply:
Good catch, can we fix this please.

Reply:
Done